### PR TITLE
[nnc][quantization] Optimize constructTensors in ext functions

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70081
* #70080
* #69954
* __->__ #69856

Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

### CODE REVIEW

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

File: torch/csrc/jit/tensorexpr/external_functions.cpp [hunk 1]
```diff
@@ -119,6 +72,110 @@ at::Tensor from_blob_quantized(
...
+                         .dtype(buf_dtypes_vec[i])
+                         .layout(at::kStrided)
+                         .device(at::kCPU) // TODO: support GPUs too
+                         .memory_format(deduce_memory_format(
+                             // NOLINTNEXTLINE
+                             buf_strides_vec[i],
+                             // NOLINTNEXTLINE
+                             buf_dims_vec[i]))
+                         .requires_grad(false);
+      if (auto qd = qdata[i]) {
```
Comment (kimishpatel) [inline torch/csrc/jit/tensorexpr/external_functions.cpp:141]: Can you separate out assign from the if for better readability?

Comment (IvanKobzarev) [reply torch/csrc/jit/tensorexpr/external_functions.cpp:141]: @kimishpatel Why does it make code less readable ? :) 
I thought the main point of introducing initialization inside the if conditions was to increase readability:
https://skebanga.github.io/if-with-initializer/

Comment (kimishpatel) [reply torch/csrc/jit/tensorexpr/external_functions.cpp:141]: I was mainly coming from "hey is this a typo?" and then I realized no it isnt. The argument in the link seems to be more from the perspective of more verbose code specifically due to the usage pattern it has. I am fine if you want to keep it this way. It is matter of personal preference.

**Review by ZolotukhinM — APPROVED** [review]
Overall looks good to me, but I'm not sure about the sigmoid change. Is there a way we can avoid it?

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

**Comment by IvanKobzarev** [comment]
@IvanKobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).
